src/Combinatorics/Matroids/properties.jl

```diff
@@ -0,0 +1,1040 @@
+        jl_flats = reverse(jl_flats)
+    end
+    matroid_flats = [[M.groundset[i] for i in flat] for flat in jl_flats]
+    if r!=nothing
+        if r<0 || r>rank(M)
+            error("The specified rank needs to be between 0 and the rank of the matroid.")
+        end
+        matroid_flats = filter(flat -> rank(M,flat)==r, matroid_flats)
+    end
+    return matroid_flats
+end
+
+@doc Markdown.doc"""
+    cyclic_flats(M::matroid, r::Union{Int,Nothing})
+
+Return the list of cyclic flats of the matroid `M`.
+These are the flats that are the union of cycles.
+See Section 2.1 in [Oxl11](@cite).
+
+By default all cylic flats are returned.
+One may specify a rank `r` as the second parameter.
+In this case only the cylic flats of this rank are returned.
+
+# Example
+```jldoctest
+julia> M = fano_matroid()
+Matroid of rank 3 on 7 elements
+
+julia> cyclic_flats(M)
+9-element Vector{Vector{T} where T}:
+ Any[]
+ [1, 2, 3]
+ [1, 4, 5]
+ [1, 6, 7]
+ [2, 4, 6]
+ [2, 5, 7]
+ [3, 5, 6]
+ [3, 4, 7]
+ [1, 2, 3, 4, 5, 6, 7]
+
+julia> cyclic_flats(M, 2)
+7-element Vector{Vector{T} where T}:
+ [1, 2, 3]
+ [1, 4, 5]
+ [1, 6, 7]
+ [2, 4, 6]
+ [2, 5, 7]
+ [3, 5, 6]
+ [3, 4, 7]
+```
+"""
+function cyclic_flats(M::Matroid, r::Union{Int,Nothing}=nothing)
+    num_flats = M.pm_matroid.LATTICE_OF_CYCLIC_FLATS.N_NODES
+    pm_flats = M.pm_matroid.LATTICE_OF_CYCLIC_FLATS.FACES
+    jl_flats = [Vector{Int}(Polymake.to_one_based_indexing(Polymake._get_entry(pm_flats, i))) for i in 0:(num_flats-1)]
+    if M.pm_matroid.LATTICE_OF_FLATS.TOP_NODE==0
+        jl_flats = reverse(jl_flats)
+    end
+    matroid_flats = [[M.groundset[i] for i in flat] for flat in jl_flats]
+    if r!=nothing
+        if r<0 || r>rank(M)
+            error("The specified rank needs to be between 0 and the rank of the matroid.")
+        end
+        matroid_flats = filter(flat -> rank(M,flat)==r, matroid_flats)
+    end
+    return matroid_flats
+end
+
+@doc Markdown.doc"""
+    closure(M::matroid, set::Vector)
+
+Return the closure of `set` in the matroid `M`.
+
+# Example
+```jldoctest
+julia> closure(fano_matroid(), [1,2])
+3-element Vector{Int64}:
+ 1
+ 2
+ 3
+```
+"""
+function closure(M::Matroid, set::GroundsetType)
+    cl = M.groundset
+    for flat in flats(M)
+        if issubset(set,flat) && issubset(flat,cl)
+            cl = flat
+        end
+    end
+    return cl
+end
+
+@doc Markdown.doc"""
+    rank(M::matroid)
+
+Return the rank of the matroid `M`.
+
+# Example
+```jldoctest
+julia> rank(fano_matroid())
+3
+```
+"""
+rank(M::Matroid) = M.pm_matroid.RANK::Int
+
+@doc Markdown.doc"""
+    rank(M::matroid, set::Vector)
+
+Return the rank of `set` in the matroid `M`.
+
+# Example
+```jldoctest
+julia> M = fano_matroid();
+
+julia> rank(M, [1,2,3])
+2
+```
+"""
+function rank(M::Matroid, set::GroundsetType)
+    if length(set)==0
+        return 0
+    else
+        return Polymake.matroid.rank( M.pm_matroid, Set([M.gs2num[i]-1 for i in set]) )
```

Comment:
Does `Polymake.matroid.rank` have a return type Julia can infer? If not, you may want to add a type assertion to get type stability (you can use `@code_warntype rank(some_matroid, some_set)` to find out whether this is actually needed or not)
```suggestion
        return Polymake.matroid.rank( M.pm_matroid, Set([M.gs2num[i]-1 for i in set]) )::Int
```

Reply:
Done